### PR TITLE
new desktop selection

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

File: src/lib/installation/custom_patterns.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+require "yast"
+
+Yast.import "PackagesUI"
+
+module Installation
+  class CustomPatterns
+    class << self
+      # flag if custom patters should be shown
```
Comment (imobachgs) [inline src/lib/installation/custom_patterns.rb]: `patterns`

File: src/lib/installation/dialogs/desktop_roles.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
+require "yast"
+require "installation/widgets/system_role"
+require "cwm/widget"
+
+Yast.import "CWM"
+# add-on product holds also online repos
+Yast.import "AddOnProduct"
+
+module Installation
+  # small widget implmenetation. Probably not worth reuse elsewhere
```
Comment (imobachgs) [inline src/lib/installation/dialogs/desktop_roles.rb]: `implementation`

File: src/lib/installation/dialogs/desktop_roles.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  class OnlineRepos < CWM::CheckBox
```
Comment (imobachgs) [inline src/lib/installation/dialogs/desktop_roles.rb]: I would move this class to a new file.

File: src/lib/installation/dialogs/desktop_roles.rb [hunk 4]
```diff
@@ -18,13 +18,25 @@ def run
...
+          caption: _("Computer Role"),
+          skip_store_for: [:redraw]
+        )
+        break if ret != :redraw
+      end
 
       Yast::Wizard.CloseDialog if separate_wizard_needed?
 
+      # support passing addon as cmd argument, openQA use it for testing
+      if !Yast::Linuxrc.InstallInf("addon").nil?
```
Comment (lslezak) [inline src/lib/installation/dialogs/desktop_roles.rb]: I prefer to use the condition without the negative operator when the `if` contains both branches.

**Pull request merged**